src/options.cpp

```diff
@@ -558,6 +552,62 @@ void options_manager::add( const std::string &sNameIn, const std::string &sPageI
     options[sNameIn] = thisOpt;
 }
 
+void options_manager::add_empty_line( const std::string &sPageIn )
+{
+    for( Page &p : pages_ ) {
+        if( p.id_ == sPageIn ) {
+            p.items_.emplace_back( ItemType::BlankLine, "", adding_to_group_ );
+            break;
+        }
+    }
+}
+
+void options_manager::add_option_group( const std::string &page_id,
+                                        const options_manager::Group &group,
+                                        const std::function<void( const std::string & )> &entries )
+{
+    if( !adding_to_group_.empty() ) {
+        // Nested groups are not allowed
+        debugmsg( "Tried to create option group '%s' from within group '%s'.",
+                  group.id_, adding_to_group_ );
+        return;
+    }
+    for( Group &g : groups_ ) {
+        if( g.id_ == group.id_ ) {
+            debugmsg( "Option group with id '%s' already exists", group.id_ );
+            return;
+        }
+    }
+    groups_.push_back( group );
+    adding_to_group_ = groups_.back().id_;
+
+    for( Page &p : pages_ ) {
+        if( p.id_ == page_id ) {
+            p.items_.emplace_back( ItemType::GroupHeader, group.id_, adding_to_group_ );
+            break;
+        }
+    }
+
+    entries( page_id );
+
+    adding_to_group_.clear();
+}
+
+const options_manager::Group &options_manager::find_group( const std::string &id ) const
+{
+    static Group null_group;
+    if( id.empty() ) {
+        return null_group;
+    }
+    for( const Group &g : groups_ ) {
+        if( g.id_ == id ) {
+            return g;
+        }
+    }
+    debugmsg( "Option group with id '%d' does not exist.", id );
```

Comment:
Just noticed, there was a type mismatch in this format string in the original PR
```suggestion
    debugmsg( "Option group with id '%s' does not exist.", id );
```